horovod/mxnet/__init__.py

```diff
@@ -43,23 +43,29 @@ def __getattr__(self, item):
     def create_state_multi_precision(self, index, weight):
         return self._optimizer.create_state_multi_precision(index, weight)
 
+    def _do_allreduce(self, index, grad):
+        if isinstance(index, (tuple, list)):
+            for i in range(len(index)):
+                allreduce_(grad[i], average=True, name=str(index[i]))
```

Comment:
Same, let's name operations `'allreduce.%d' % index[i]` for clarity?

Reply:
We already prefix the index in mpiops.cc. See [here](https://github.com/uber/horovod/blob/master/horovod/mxnet/mpi_ops.cc#L186)